sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -918,4 +1043,26 @@ public static void ResetFeatureSwitch()
                 "AZURE_EXPERIMENTAL_ENABLE_ACTIVITY_SOURCE");
         }
     }
+#else
+#pragma warning disable SA1507 // File can not contain multiple types
+    /// <summary>
+    /// Until Activity Source is no longer considered experimental.
```

Comment:
Curious: why is ActivitySource considered experimental, and what needs to change to make it not that?

Reply:
I think @lmolkova would probably be able to answer this better than I can. I'm not entirely sure!

Reply:
The ActivitySource itself can be freely used - the OTel support however is experimental. OTel support requires us to add attributes to the activity (e.g. `http.url`) to describe HTTP and other requests. 

Attribute semantics are still experimental and subject to change (they already did and we're yet to update them). Latest version of OTel HTTP semantics is already frozen, and we'll be dropping the feature flag once they become stable (likely within next few months).